selftests/unit/test_loader.py

```diff
@@ -401,6 +400,76 @@ def test_load_tags(self):
                 del(tags_map[name])
         self.assertEqual(len(tags_map), 0)
 
+    def test_filter_tags_ignore_empty(self):
+        avocado_pass_test = script.TemporaryScript('passtest.py',
+                                                   AVOCADO_TEST_OK,
+                                                   'avocado_loader_unittest',
+                                                   DEFAULT_NON_EXEC_MODE)
+        with avocado_pass_test as test:
+            test_suite = self.loader.discover(test.path, loader.ALL)
+            self.assertEqual([], loader.filter_test_tags(test_suite, []))
+            self.assertEqual(test_suite,
+                             loader.filter_test_tags(test_suite, [], True))
+
+    def test_filter_tags(self):
+        avocado_test_tags = script.TemporaryScript('tags.py',
+                                                   AVOCADO_TEST_TAGS,
+                                                   'avocado_loader_unittest',
+                                                   DEFAULT_NON_EXEC_MODE)
+        with avocado_test_tags as test:
+            test_suite = self.loader.discover(test.path, loader.ALL)
+            self.assertEqual(len(test_suite), 5)
+            self.assertEqual(test_suite[0][0], 'SafeTest')
+            self.assertEqual(test_suite[0][1]['methodName'], 'test_safe')
```

Comment:
I thought about it, but since this is indeed a series of match/checks, it seems clearer to do it directly than to add a intermediate function.  What I'm trying to say is, I chose explicitness over compactness.

I guess It's a matter of preference since you think the current style doesn't make for good code readability...